scripts/eqtl_hail_batch/conditional_analysis.py

```diff
@@ -0,0 +1,528 @@
+    gene_snp_test_df = esnps_to_test[['snpid', 'gene_symbol', 'gene_id']]
+    gene_snp_test_df = gene_snp_test_df[
+        gene_snp_test_df['gene_symbol'] == gene_ids.iloc[idx]
+    ]
+    # Subset genotype file for the significant SNPs
+    genotype_df = get_genotype_df(
+        filtered_mt_path, residual_df, gene_snp_test_df
+    )
+
+    def spearman_correlation(df):
+        """get Spearman rank correlation"""
+        gene_symbol = df.gene_symbol
+        gene_id = df.gene_id
+        snp = df.snpid
+        gt = genotype_df[genotype_df.snpid == snp][['sampleid', 'n_alt_alleles']]
+
+        res_val = residual_df[['sampleid', gene_symbol]]
+        test_df = res_val.merge(gt, on='sampleid', how='right')
+        test_df.columns = ['sampleid', 'residual', 'SNP']
+        spearmans_rho, p = spearmanr(
+            test_df['SNP'], test_df['residual'], nan_policy='omit'
+        )
+        return (gene_symbol, gene_id, snp, spearmans_rho, p)
+
+    # calculate spearman correlation
+    adjusted_spearman_df = pd.DataFrame(
+        list(gene_snp_test_df.apply(spearman_correlation, axis=1))
+    )
+    adjusted_spearman_df.columns = [
+        'gene_symbol',
+        'gene_id',
+        'snpid',
+        'spearmans_rho',
+        'p_value',
+    ]
+    # remove any NA values. Remove this line when run on main dataset
+    adjusted_spearman_df = adjusted_spearman_df.dropna(axis=0, how='any')
+    # add in global position and round
+    locus = adjusted_spearman_df.snpid.str.split(':', expand=True)[[0, 1]].agg(
+        ':'.join, axis=1
+    )
+    chrom = locus.str.split(':', expand=True)[0]
+    bp = locus.str.split(':', expand=True)[1]
+    (
+        adjusted_spearman_df['locus'],
+        adjusted_spearman_df['chrom'],
+        adjusted_spearman_df['bp'],
+    ) = [
+        locus,
+        chrom,
+        bp,
+    ]
+    adjusted_spearman_df['round'] = iteration
+    t = hl.Table.from_pandas(adjusted_spearman_df)
+    t = t.annotate(global_bp=hl.locus(t.chrom, hl.int32(t.bp)).global_position())
+    t = t.annotate(locus=hl.locus(t.chrom, hl.int32(t.bp)))
+    # get alleles
+    mt = hl.read_matrix_table(filtered_mt_path).key_rows_by('locus')
+    t = t.key_by('locus')
+    t = t.annotate(
+        a1=mt.rows()[t.locus].alleles[0],
+        a2=mt.rows()[t.locus].alleles[1],
+    )
+    t = t.annotate(
+        id=hl.str(':').join(
+            [
+                hl.str(t.chrom),
+                hl.str(t.bp),
+                t.a1,
+                t.a2,
+                t.gene_symbol,
+                # result.db_key, # cell_type_id (eg nk, mononc)
+                hl.str(t.round),
+            ]
+        )
+    )
+    adjusted_spearman_df = t.to_pandas()
+
+    # set variables for next iteration of loop
+    significant_snps_df = adjusted_spearman_df
+
+    return significant_snps_df
+
+
+def merge_significant_snps_dfs(*df_list):
+    """
+    Merge list of list of sig_snps dataframes
+    """
+
+    merged_sig_snps = pd.concat(df_list)
```

Comment:
Import multipy here instead (this will give you a pylint error which you can suppress) 

```suggestion
    from multipy.fdr import qvalue 
    merged_sig_snps = pd.concat(df_list)
```